src/test/java/org/opensearch/sdk/TestBaseExtension.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.sdk;
+
+import org.junit.jupiter.api.Test;
+import org.opensearch.test.OpenSearchTestCase;
+
+import java.util.List;
+
+public class TestBaseExtension extends OpenSearchTestCase {
+
+    private static final String EXTENSION_DESCRIPTOR_CLASSPATH = "/bad-extension.yml";
+    private static final String EXTENSION_DESCRIPTOR_FILEPATH = "src/test/resources" + EXTENSION_DESCRIPTOR_CLASSPATH;
+
+    private class TestExtension extends BaseExtension {
+
+        protected TestExtension(String path) {
+            super(path);
+        }
+
+        @Override
+        public List<ExtensionRestHandler> getExtensionRestHandlers() {
+            return null;
```

Comment:
Doesn't `BaseExtension` give us a default empty list?  What value is there overriding here with null?

Reply:
Looks like it doesn't.

```
    /**
     * Gets a list of {@link ExtensionRestHandler} implementations this extension handles.
     *
     * @return a list of REST handlers (REST actions) this extension handles.
     */
    List<ExtensionRestHandler> getExtensionRestHandlers();
```
We could change it to have a default implementation in the `Extension.java` interface.

Reply:
Yes, the `ActionPlugin` interface we are mimicking does so (along with some extension points!)
```
    /**
     * Rest handlers added by this plugin.
     */
    default List<RestHandler> getRestHandlers(
        Settings settings,
        RestController restController,
        ClusterSettings clusterSettings,
        IndexScopedSettings indexScopedSettings,
        SettingsFilter settingsFilter,
        IndexNameExpressionResolver indexNameExpressionResolver,
        Supplier<DiscoveryNodes> nodesInCluster
    ) {
        return Collections.emptyList();
    }
```